core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponentSpec.scala

```diff
@@ -239,6 +240,72 @@ class SubmissionComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
     }
   }
 
+  it should "return workflow failure messages even when workflowRec.workflowEntityId is None" in withDefaultTestDatabase {
+    // example submission, no workflows as of yet
+    val sid = UUID.randomUUID()
+    val submissionNoEntitiesYesMessages = Submission(
+      submissionId = sid.toString,
+      submissionDate = testDate,
+      submitter = WorkbenchEmail(testData.userOwner.userEmail.value),
+      methodConfigurationNamespace = testData.methodConfigValid.namespace,
+      methodConfigurationName = testData.methodConfigValid.name,submissionEntity = None,
+      workflows = Seq(),
+      status = SubmissionStatuses.Done,
+      useCallCache = false,
+      deleteIntermediateOutputFiles = false
+    )
+
+    withWorkspaceContext(testData.workspace) { context =>
+      // save the submission
+      runAndWait(DBIO.seq(
+        submissionQuery.create(context, submissionNoEntitiesYesMessages),
+        updateWorkflowExecutionServiceKey("unittestdefault") ))
+
+      // save a couple workflow rows to this submission.
+      val workflows = Seq(
+        WorkflowRecord(11111, Option("workflow1"), sid, WorkflowStatuses.Failed.toString, Timestamp.from(java.time.Instant.now()),
+          None, 1, Option("unittestdefault"), None),
+        WorkflowRecord(22222, Option("workflow2"), sid, WorkflowStatuses.Failed.toString, Timestamp.from(java.time.Instant.now()),
+          None, 1, Option("unittestdefault"), None),
+      )
+      val saveWorkflows = workflowQuery ++= workflows
+      runAndWait(saveWorkflows)
+      val actualWorkflows = runAndWait(workflowQuery.findWorkflowsBySubmissionId(sid).result)
+      withClue("checking count of saved workflows from findWorkflowsBySubmissionId: ") {
+        actualWorkflows.size shouldBe 2
+      }
```

Comment:
I don't know what use cases cause this particular shape of data to land in the database - `WorkflowRecord.workflowEntityId` is `None` - and I couldn't easily find a higher-level method to insert that data, so I'm just inserting the raw rows.